expression/builtin_cast_vec.go

```diff
@@ -756,11 +756,45 @@ func (b *builtinCastStringAsJSONSig) vecEvalJSON(input *chunk.Chunk, result *chu
 }
 
 func (b *builtinCastRealAsDecimalSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastRealAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err = b.args[0].VecEvalDecimal(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.ResizeDecimal(n, false)
+	result.MergeNulls(buf)
+	bufreal := buf.Float64s()
+	resdecimal := result.Decimals()
+	if !b.inUnion {
+		for i := 0; i < n; i++ {
+			if err = resdecimal[i].FromFloat64(bufreal[i]); err != nil {
+				return err
+			}
+		}
+	} else {
+		for i := 0; i < n; i++ {
+			if bufreal[i] >= 0 {
+				if err = resdecimal[i].FromFloat64(bufreal[i]); err != nil {
+					return err
+				}
+
+			} else {
+				_, err = types.ProduceDecWithSpecifiedTp(&resdecimal[i], b.tp, b.ctx.GetSessionVars().StmtCtx)
```

Comment:
In non-vec function, this line will run in all time whatever b.inUnion and val is (of course, without error). But in vec function (your code),  it only runs when b.inUnion is true and bufreal[i] < 0.

Reply:
ok ,I will fix these two problem as soon as possible